lib/errors.js

```diff
@@ -9,18 +9,48 @@ const codes = {
   INVALID_DELETION_OPERATION: 1004,
   INVALID_CREATION_OPERATION: 1005,
   INSUFFICIENT_PERMISSIONS: 1006,
+  ACTION_EXECUTION_ERROR: 1007,
 };
 
 const defaultMessages = {
   [codes.NOT_IMPLEMENTED]: 'Not implemented',
 };
 
+class ActionError extends Error {
+  constructor(name, ctx) {
+    super(name);
+
+    this.name = 'ActionError';
+    this.ctx = ctx;
```

Comment:
The `codes` are for `GSError` only, not for `ActionError`. Also, there is no need in doing that, since it can already be checked for as `error instanceof ActionError`.